rust/src/builder.rs

```diff
@@ -389,7 +389,10 @@ mod tests {
         assert_eq!("memory://", uri.as_str());
         let uri = ensure_table_uri("s3://tests/data/delta-0.8.0/").unwrap();
         assert_eq!("s3://tests/data/delta-0.8.0", uri.as_str());
-        let _uri = ensure_table_uri("s3://tests/data/delta-0.8.0//").unwrap();
-        assert_eq!("s3://tests/data/delta-0.8.0", uri.as_str())
+        let uri = ensure_table_uri("s3://tests/data/delta-0.8.0//").unwrap();
+        assert_eq!("s3://tests/data/delta-0.8.0", uri.as_str());
+
+        let uri = ensure_table_uri("file://test path/").unwrap();
+        assert_eq!("file://test path/", uri.as_str());
```

Comment:
@wjones127 These URIs are not valid file URLs to begin with and I believe that the test failure is a misnomer. According to [very official and always correct sources](https://en.wikipedia.org/wiki/File_URI_scheme#Unix) three slashes are required.

I checked this topic branch out, merged the latest `main` with it and ran the tests with three slashes.  The assertion does fail when I use `file:///tmp/test path`

```
thread 'builder::tests::test_ensure_table_uri' panicked at 'assertion failed: `(left == right)`
  left: `"file:///tmp/test path/"`,
 right: `"file:///tmp/test%20path"`', rust/src/builder.rs:450:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

_However_, the `/tmp/test path/` directory does exist correctly on the file system, having been created by the code which lives in `ensure_table_uri`.

I can push my merge of `main` if you'd like.

I think this test _may_ not be correct and we may have fixed the spaces-in-table-names issues since the rest of the assertions pass with the latest `HEAD` of `main`